### PR TITLE
boards: fix dependency for LV_COLOR_SWAP_16 to use configdefault

### LINKED ISSUE #81546: `LV_COLOR_SWAP_16` has incorrect dependencies across Zephyr
**Describe the bug**
The KConfig `LV_COLOR_SWAP_16` is defined to depend on `LV_COLOR_DEPTH_16` within LVGL itself: https://github.com/zephyrproject-rtos/lvgl/blob/a76caad0aa99838bb7a3315aaaf07e50d595c284/Kconfig#L40

However, many shields or boards in Zephyr simply enable this setting if LVGL is enabled, since they assume LVGL will be built with `LV_COLOR_DEPTH_16` when targeting that shield/board. For example, see here: https://github.com/zephyrproject-rtos/zephyr/blob/4789820722f36d523830daa540524ef0a4230dce/boards/weact/mini_stm32h743/Kconfig.defconfig#L16

This assumption is broken if a sample or user  manually selects the LVGL color depth setting, such as in the transparency sample: https://github.com/zephyrproject-rtos/zephyr/blob/4789820722f36d523830daa540524ef0a4230dce/samples/modules/lvgl/screen_transparency/prj.conf#L6

This results in a build error, since `LV_COLOR_DEPTH_16` should not logically be enabled.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? `m5stack_atoms3/esp32s3/procpu`

**To Reproduce**
1. `west build -p -b m5stack_atoms3/esp32s3/procpu samples/modules/lvgl/screen_transparency -T sample.modules.lvgl.screen_transparency`

**Expected behavior**

Setting the choice Kconfig `LV_COLOR_DEPTH` should be possible without inducing a build error.

**Impact**
Blocking CI on #81293

**Logs and console output**
```
n.c.obj -c /home/danieldegrasse/zephyr/zephyrproject/zephyr/samples/modules/lvgl/screen_transparency/src/main.c
In file included from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../src/hal/../draw/../misc/lv_style.h:19,
                 from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../src/hal/../draw/lv_draw.h:18,
                 from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../src/hal/lv_hal_disp.h:21,
                 from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../src/hal/lv_hal.h:16,
                 from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../lvgl.h:33,
                 from /home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/lvgl.h:17,
                 from /home/danieldegrasse/zephyr/zephyrproject/zephyr/samples/modules/lvgl/screen_transparency/src/main.c:11:
/home/danieldegrasse/zephyr/zephyrproject/modules/lib/gui/lvgl/src/../src/hal/../draw/../misc/lv_color.h:27:2: error: #error "LV_COLOR_16_SWAP requires LV_COLOR_DEPTH == 16. Set it in lv_conf.h"
   27 | #error "LV_COLOR_16_SWAP requires LV_COLOR_DEPTH == 16. Set it in lv_conf.h"
      |  ^~~~~
[30/433] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/rb.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/danieldegrasse/zephyr/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**

 - OS: Ubuntu 22.04 LTS
 - Toolchain Zephyr SDK 0.17.0-rc1
 - Commit SHA or Version used 4789820722f36d (main)

### PR DESCRIPTION
**Pull request by danieldegrasse**
Use configdefault when enabling LV_COLOR_SWAP_16 within boards and
shield definitions, to avoid OR'ing the dependencies for the Kconfig
symbol. Otherwise, a user manually selecting LV_COLOR_DEPTH will
encounter build errors as LV_COLOR_SWAP_16 may be enabled when
LV_COLOR_DEPTH_16 is not selected

Fixes #81546

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@faxe1008 do you feel this is the right fix? Or should the LVGL transparency sample not be setting `CONFIG_LV_COLOR_DEPTH_32` directly?

**Review by kartben — DISMISSED** [review]

**Review by uLipe — DISMISSED** [review]

**Review by wmrsouza — DISMISSED** [review]

**Review by faxe1008 — COMMENTED** [review]
Hmm wouldn't the easier fix be adding the depedency to the stub symbol within `modules/lvgl/Kconfig`?

EDIT:// yeah you are right, this wont suffice since they are ORed. Regarding the sample, selecting the depth makes sense since otherwise the sample is kinda pointless.

**Review by faxe1008 — APPROVED** [review]

File: boards/adi/max32662evkit/Kconfig.defconfig [hunk 1]
```diff
@@ -20,7 +20,7 @@ choice LV_COLOR_DEPTH
 endchoice
 
 config LV_COLOR_16_SWAP
-	default y
+	default y if LV_COLOR_DEPTH_16
```
Comment (erwango) [inline boards/adi/max32662evkit/Kconfig.defconfig:23]: what about using [`configdefault`](https://docs.zephyrproject.org/latest/build/kconfig/extensions.html#kconfig-extensions) ?

Comment (faxe1008) [reply boards/adi/max32662evkit/Kconfig.defconfig:23]: Very good call, forgot that this exists

**Review by faxe1008 — CHANGES_REQUESTED** [review]
See @erwango comment

File: boards/adi/max32662evkit/Kconfig.defconfig (hunk 1 above)
Comment (danieldegrasse) [reply boards/adi/max32662evkit/Kconfig.defconfig:23]: Wasn't aware of this- I'll update the PR to use this, thanks!

**Comment by danieldegrasse** [comment]
> See @erwango comment

Updated

**Comment by danieldegrasse** [comment]
#81611 should fix the other test failure blocking this PR

**Review by kartben — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**